libcontainer/dmz/cloned_binary_linux.go

```diff
@@ -19,6 +20,23 @@ var (
 	_ SealFunc = sealFile
 )
 
+func isExecutable(f *os.File) bool {
+	if err := unix.Faccessat(int(f.Fd()), "", unix.X_OK, unix.AT_EACCESS|unix.AT_EMPTY_PATH); err == nil {
+		return true
+	} else if errors.Is(err, unix.EACCES) {
```

Comment:
Same here, we can use direct err comparison (see https://github.com/opencontainers/runc/pull/3995).